sunpy/net/jsoc/jsoc.py

```diff
@@ -518,9 +522,17 @@ def _make_query_payload(self, start_time, end_time, series, notify=None,
                    'requestor': 'none',
                    'filenamefmt': '{0}.{{T_REC:A}}.{{CAMERA}}.{{segment}}'.format(series)}
 
-        payload.update(kwargs)
+        payload.update(self._clean_primarykey(kwargs))
         return payload
 
+    def _clean_primarykey(self, kwargs):
```

Comment:
provides a dictionary without the 'primarykey_' in the keyword...

So, what I'm doing is `jsoc.attr.PrimaryKey(a,b)` is translated later on into a dictionary.  to differentiate it from the others I'm making their keyword to contain 'primarykey_'  such as:  `{primarykey_a:b}`.  Then this ` _clean_primarykey()` just removes that bit for the queries.
All that comes because `_make_record_set` doesn't uses all the kewyords, just series, wavelength, segment and primaryKeys... (also the `[? xxx ~ yyyy ?]` sql queries that I've not figured out yet how to allow them). And the only way I've found to separates the ones to download (like Notify) from the ones to query is by using `primarykey_a` in the dictionary's key and then remove it ([like in line 490](https://github.com/sunpy/sunpy/pull/1400/files#diff-61f5137ce8bef90efb1dd91756542acbR490))...

If you have a better idea, please let me know it... I think this is a bit patchy, but I don't know how to transfer the information of this new class.